pypiserver/plugin.py

```diff
@@ -0,0 +1,88 @@
+"""
+===============================
+Setup-tools Plugin Architecture
+===============================
+
+`setup.py` configurations
+-------------------------
+To implement a new plugin, you have to package your code as a regular
+python distribution and add the following declaration inside its
+:file:`setup.py`::
+
+    setup(
+        # ...
+        entry_points = {
+            'pypiserver.plugins': [
+                'plugin_1 = <foo.plugin.module>:<plugin-install-func> ## Load & install.
+                'plugin_2 = <bar.plugin.module> [dep1, dep2]          ## Load only.
+            ]
+        }
+    )
+
+
+Implementing a plugin
+---------------------
+The plugins are initialized alphabetically in a 2-stage procedure
+during *import time* by :func:`init_plugins()`:
+
+- A plugin is *loaded* and
+- optionally *installed* if the configuration in `setup.py ` specifies
+  a no-args ``<plugin-install-func>``.
+
+Any collected ``<plugin-install-func>`` callables are invoked AFTER all
+plugin-modules have finished loading.
+
+.. Warning::
+   When appending into "hook" lists during installation, remember to avoid
+   re-inserting duplicate items.  In general try to well-behave even when
+   **plugins are initialized multiple times**!
+
+"""
+from _collections import OrderedDict
```

Comment:
What's the reason for using the `_collections` extension module rather than `collections`?

Reply:
Oups! thanks